src/transformers/agents/tools.py

```diff
@@ -349,96 +385,103 @@ def __init__(self, _gradio_tool):
                 super().__init__()
                 self.name = _gradio_tool.name
                 self.description = _gradio_tool.description
+                self.output_type = str
+                self._gradio_tool = _gradio_tool
+
+            def __call__(self, *args, **kwargs):
+                return self._gradio_tool.run(*args, *list(kwargs.values()))
 
-        GradioToolWrapper.__call__ = gradio_tool.run
         return GradioToolWrapper(gradio_tool)
 
+    @staticmethod
+    def from_langchain(langchain_tool):
+        """
+        Creates a [`Tool`] from a langchain tool.
+        """
 
-class RemoteTool(Tool):
-    """
-    A [`Tool`] that will make requests to an inference endpoint.
+        class LangChainToolWrapper(Tool):
+            def __init__(self, _langchain_tool):
+                super().__init__()
+                self.name = _langchain_tool.name
+                self.description = _langchain_tool.description
+                self.inputs = parse_langchain_args(_langchain_tool.args)
+                self.output_type = str
+                self.langchain_tool = _langchain_tool
 
-    Args:
-        endpoint_url (`str`, *optional*):
-            The url of the endpoint to use.
-        token (`str`, *optional*):
-            The token to use as HTTP bearer authorization for remote files. If unset, will use the token generated when
-            running `huggingface-cli login` (stored in `~/.huggingface`).
-        tool_class (`type`, *optional*):
-            The corresponding `tool_class` if this is a remote version of an existing tool. Will help determine when
-            the output should be converted to another type (like images).
-    """
+            def __call__(self, *args, **kwargs):
+                # This lets the user type args either as kwargs or positional arguments
+                for index, argument in enumerate(args):
+                    if index < len(self.inputs):
+                        input_key = next(iter(self.inputs))
+                        kwargs[input_key] = argument
 
-    def __init__(self, endpoint_url=None, token=None, tool_class=None):
-        self.endpoint_url = endpoint_url
-        self.client = EndpointClient(endpoint_url, token=token)
-        self.tool_class = tool_class
+                tool_input = {}
+                for key, value in kwargs.items():
+                    tool_input[key] = value
```

Comment:
We need to convert the inputs as a dict, since Langchain expects a single dictionarry as input, like `langchain_tool.run({'query': 'I want to find the nearest swimming pool'})`. but since for positional inputs we do not know the associated keyword, we need to find it, thus the additional code.

Reply:
I simplified the code here!